paper-server/src/main/java/org/bukkit/craftbukkit/entity/CraftFishHook.java

```diff
@@ -233,5 +238,11 @@ public void resetFishingState() {
         hook.resetTimeUntilLured();
         hook.timeUntilHooked = 0; // Reset time until hooked, will be repopulated once lured time is ticked down.
     }
-    // Paper end
+
+    @Override
+    public int retrieve(EquipmentSlot slot, ItemStack itemStack) {
+        Preconditions.checkNotNull(slot, "Equipment slot cannot be null");
+        Preconditions.checkNotNull(itemStack, "ItemStack cannot be null");
```

Comment:
```suggestion
        Preconditions.checkArgument(slot != null, "Equipment slot cannot be null");
        Preconditions.checkArgument(itemStack, != null "ItemStack cannot be null");
```

Reply:
Curious ... why?
Internally they do about the same thing.

Reply:
Hi... well if remember the idea is throw IllegalArgumentException (many methods from Preconditions throw that) but checkNotNull throws NPE.
and now i remember was mentioned in [#paper-contrib](https://canary.discord.com/channels/289587909051416579/925530366192779286/1336746860186173523)

Reply:
Duly noted, thank you!